src/combineEpics.ts

```diff
@@ -1,10 +1,17 @@
+import { Action } from 'redux';
 import { merge } from 'rxjs';
+import { Epic } from './epic';
 
 /**
   Merges all epics into a single one.
  */
-export const combineEpics = (...epics) => {
-  const merger = (...args) => merge(
+export function combineEpics<T extends Action, O extends T = T, S = void, D = any>(...epics: Epic<T, O, S, D>[]): Epic<T, O, S, D>;
+// TODO: Maybe we should make sure `E` is callable? (`E extends () => ...`)
+export function combineEpics<E>(...epics: E[]): E;
```

Comment:
That would be an improvement, but feel free to punt on that for this PR if you'd like. Either way.

Reply:
Since you mentioned this will make semver major, I think it's a good idea to add a minimal constraint here: `<E extends () => Observable<any>>`

Reply:
Sounds good. I'm not entirely sure why this overload is needed. It might be, or it might only exist as a remnant over the years. It would seem that the first signature, with Epic<>, would work in any valid case wouldn't it? The only argument I can see is that the second overload is far less strict, so it would accept functions that aren't yet typed 100% as epics, which I'm not sure is a good thing. If someone wants that, they can always cast to any. Thoughts?

Reply:
I checked the blame for the line. It was added [in this PR](https://github.com/redux-observable/redux-observable/commit/88c0bf9e9bd44feeed1af18ec7e5639890400e72) that mentions dependency injection, but this seems to be before the `Dependencies` generic was added to the `Epic` interface. Could be this is not necessary anymore due to being able to use:
```ts
createEpicMiddleware({ dependencies: myDeps });
```

I'll remove this overload too. I can always revert otherwise.

Reply:
I removed all overloads for `combineEpics` in last commit.